presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java

```diff
@@ -328,6 +328,22 @@ public void testSort()
     }
 
     @Test
+    public void testArrayIntersect()
+            throws Exception
+    {
+        assertFunction("ARRAY_INTERSECT(ARRAY [12], ARRAY [10])", new ArrayType(BIGINT), ImmutableList.of());
+        assertFunction("ARRAY_INTERSECT(ARRAY ['foo', 'bar', 'baz'], ARRAY ['foo', 'test', 'bar'])", new ArrayType(VARCHAR), ImmutableList.of("bar", "foo"));
+        assertFunction("ARRAY_INTERSECT(ARRAY [NULL], ARRAY [NULL, NULL])", new ArrayType(UNKNOWN), ImmutableList.of());
```

Comment:
This test isn't correct. The behavior should be like the SQL INTERSECT construct, so the result of this should be `array[null]`